Cataclysm/Warrior.lua

```diff
@@ -1861,11 +1789,11 @@ spec:RegisterAbilities( {
     },
 
 
-    -- Reduces all damage taken by 60% for 12 sec.
+   -- Reduces all damage taken by 40% for 12 sec.
     shield_wall = {
         id = 871,
         cast = 0,
-        cooldown = function() return ( glyph.shield_wall.enabled and 180 or 300 ) - 30 * talent.improved_disciplines.rank end,
+        cooldown = function() return ( glyph.shield_wall.enabled and 180 or 300 ) end,
```

Comment:
Maybe your intent is to reduce the CD based on Shield Mastery talent being present?  If so, you'd just update the cooldown to calculate here:

`cooldown = function() return 300 - ( glyph.shield_wall.enabled and 180 or 0 ) - ( talent.shield_mastery.enabled and 180 or 0 ) end,`